### DISCUSSION TITLE
Dead Intellitouch

### DISCUSSION THREAD

**Discussion by arrmo** [tagyoureit/nodejs-poolController#575] [Q&A] [answered]
Hi!

OK, have been using njsPC for a long time 😆. But I can't now, as I have a "slight" issue - slight being a dead IntelliTouch it seems. Apologies for the pester here, but not sure where to turn, as I'm getting a bit of a smelly story from pool repair companies. If anyone doesn't mind helping a bit with troubleshooting (or even just random thoughts of things to check) ... please ping me. Would appreciate any thoughts or suggestions you may have. I can provide more details of what I am seeing and have tried, but don't want to spam everyone.

Thanks in advance!

**Comment by rstrouse** [comment, marked as answer]
Honestly,  if years ago when I had IntelliTouch the mobo died one more time (over the years it happened 3 times) and Nixie existed,  I would have simply converted it to Nixie.  All you need to do is add as many 8 relay cards to your pi for $45 each and drive the whole thing from Nixie.  It will completely replace IntelliTouch with more functionality and greater responsiveness.

Now if you are dead set on fixing it I know a guy who can fix it.  

Common Failures
* RS485 chips go bad - Schedules still work but pumps, chlorinator, remotes are dead
* Darlington transistor chips failed - IntelliTouch cannot switch relays but it still appears to work with the remotes. (you can replace them as these are the 3 socketed chips on the mobo)
* No power up - Check the power supply under load to the board.  There are some linear power regulators on the board that may need to be replaced.

**Reply by arrmo** [reply]
Thanks! Sadly, I do want to repair it - long list of (annoying) reasons ... 🤣.

Sort of no power up. What I mean is - no LED's visible. Transformer outputs look to be there, but not sure about under load. The indoor controller does light up (actually, never turns off), shows 12 am, 0F temperature ... and when I try to do almost anything, "There was an error retrieving the data". Actually wondering about a re-flash (poking around), but not sure it's easy to find an adapter (and seems you can't re-flash over RS485)? I can't talk to either RS-485 interface (indoor controller, njsPC).

I did try to talk to the personality board only, but seems you have two have the mainboard connected for that to come up as well (i.e. can't try to isolate the boards ... arrgh!).

Thoughts?

Thanks!

**Reply by rstrouse** [reply]
You will not get the personality board to do anything and in fact it does not have a failure mode that will bring the mobo down.  It sounds like you have at least 10v on the ICP red/black wire so it is getting power.  It will remain lit until while it is asking for data from the main OCP. This is normal and your ICP is waiting for the first message that contains the time.

Look at the mobo very carefully and see if you see any burned components.  Look for discoloration on both sides of the board.  A re-flash will likely do nothing for you.  If you see any post a picture here.

Also you can contact Tom via his blog https://saturationregion.blogspot.com/

**Reply by arrmo** [reply]
> Look at the mobo very carefully and see if you see any burned components. Look for discoloration on both sides of the board. A re-flash will likely do nothing for you. If you see any post a picture here.

Will do, thanks! Assuming you're meaning linear regulators and/or switchers (and associated FET's)? Will take a look - was off at 🏒  tonight 😄.

**Reply by rstrouse** [reply]
Yes that would b a likely component given the fact that it has power but not booting.  105 here today so hockey looks a lot like wading in skates.

**Reply by arrmo** [reply]
LOL - ya, ice was a bit soft 😆. I'm near D/FW, TX ... so pretty warm here also.

**Reply by arrmo** [reply]
OK, thinking this may be the part - U10. Plastic is a bit bubbled, typical of a blown part. Agreed? Arrgh, why won't the picture upload? Tried jpg, png ... sigh. Or do you see it?

Having an issue reading the part number on it though. Do you happen to know? And what is it?

Thanks!
![IMG_0696](https://user-images.githubusercontent.com/8756384/185755883-6e94647d-a458-444a-b6d9-90c2c6e31a7f.png)

**Reply by arrmo** [reply]
Let me try this - zipped.
[IMG_0696.zip](https://github.com/tagyoureit/nodejs-poolController/files/9387641/IMG_0696.zip)

**Reply by rstrouse** [reply]
I do not see that image but a bubbled part is typically a good sign or a bad sign depending on which direction you fill a glass from.

**Reply by rstrouse** [reply]
To me that looks like a 74HC244D.

**Reply by rstrouse** [reply]
I also see a little bit of damage on R69.   Look closely at the capacitors and components near U17.  Are any of these caps swollen or the regulator discolored.

**Reply by arrmo** [reply]
Dang, good eye! I missed R69, but I see it now - looks like discolored flux => typically from running hot.
 
Not seeing any swollen electrolytics around U17 (attaching the zip here). There is a bit of "residue" on the body of it ... but that doesn't look too bad to me (feel free to disagree!). Leads are perhaps slightly discolored.

Do you know the voltages that should be there? I can probe around, try to see what they look like.

Thanks!
[IMG_0697.zip](https://github.com/tagyoureit/nodejs-poolController/files/9387740/IMG_0697.zip)

**Reply by rstrouse** [reply]
Sorry that jpg is not showing for me.  I can only assume that it is some sort of crapple image extension.   I don't know what the voltages should be there.

**Reply by arrmo** [reply]
Arrgh! No idea what is going on with the pics. Very odd. Let me open the pic, snip, paste here, this work?
![image](https://user-images.githubusercontent.com/8756384/185760298-fa14e5df-1a8b-40f8-a1f6-e9723c0e51a7.png)

Seems to, from preview 🤣.

OK, plugged it back together, slightly odd ... LOL. + side of C4, ~ 17 VDC (input to L1 I think), But the output of L1, input to the regulator => ~ 0.24V ... and the regulator output is nada (~ 0V). So fried regulator, even though it's not looking so bad?

Thanks!

**Reply by rstrouse** [reply]
I think 17v is off the tap of the transformer.  I would have expected 5v out of the regulator to power the chip.

**Reply by arrmo** [reply]
Yep, agreed! So looks like transformer output is OK (it's not being dragged down at C4 output, but there is no input to the regulator => regulator gone, and took out (fused) L1 in the process?

**Reply by rstrouse** [reply]
Not horrible chances that you have identified the all components that need to be changed.

**Reply by arrmo** [reply]
Thanks! Do you happen to know what L1 is (inductor? What value?)?

**Reply by rstrouse** [reply]
It is an inductor but I have no idea what value that is.

**Reply by arrmo** [reply]
NP, thanks for all the help! Out of town this week, but parts due in by next weekend 😄.

Not sure how to do this in GitHub, but can I PM you with a question?

Thanks again.

**Reply by rstrouse** [reply]
You can PM me on gitter.  I don't know why that isn't available in discussions.

**Reply by arrmo** [reply]
FYI, it hit me as I was flying out for last week ... the input node to the regulator seemed to be floating, so when I got back => checked L1, it's high impedance. Tried a short around it (just to test) ... more smoke from the hex driver. I think the regulator may be OK, it's the hex driver that went, and took out L1. Not sure what to do about L1 - just short around it after, once the driver is replaced?

Thanks!

**Reply by rstrouse** [reply]
It seems so odd that the inductor would be bad and there would be no visible damage.  Is there no continuity through the inductor?

**Reply by arrmo** [reply]
Agreed! No, at least the resistance through it is quite high. Sort of hit me after I had been out there, as the input sagged as the DVM was on it (measuring input to the regulator) => sign of high drive impedance. 

Make sense?

**Reply by rstrouse** [reply]
Yeah you have just exceeded my abilities.  I do believe that a functioning inductor should have very low resistance through the coil   Given how they are constructed I couldn't imagine that it would not heat up that coil and burn something on failure.   Sadly I do not know how to determine what the values are when it isn't marked in any way.   Perhaps the values are actually on the bottom of the part.  Maybe desolder it from the board and see if there are numbers or dots under it.

If I were to guess though I would say this.  But it is just a guess.
https://www.digikey.com/en/products/detail/murata-electronics/LQH43MN102K03L/3845851?utm_adgroup=Fixed%20Inductors&utm_source=google&utm_medium=cpc&utm_campaign=Shopping_Product_Inductors%2C%20Coils%2C%20Chokes&utm_term=&utm_content=Fixed%20Inductors&gclid=EAIaIQobChMIgOnU48vs-QIV1j-tBh3wxgRMEAYYByABEgK96vD_BwE

**Comment by arrmo** [comment]
Yes, that does look quite similar, thanks! For now, will jumper around it, as the cap and inductor are really just a LPF (for noise rejection). Let's see 😆 

Thanks again.

**Reply by rstrouse** [reply]
My guess ripple reduction for the comm ports

**Reply by arrmo** [reply]
I could be wrong, but as it's on the input side of the regulator, and out of the caps - thinking inrush current limiting, but I may be wrong.

And ... fixed! It was the hex driver => died, regulator still OK, inductor fused. Replaced the hex driver, and inductor, all up again. Thanks for the help!!!

**Reply by rstrouse** [reply]
Awesome!

**Reply by tagyoureit** [reply]
Dang!  Pretty impressive...  Nice job.

**Reply by arrmo** [reply]
Thanks! Really appreciate all the pointers and help.

Send folks my way next time you hear of some like this - learned quite a bit, would like to "pay it forward" if I can!

**Comment by arrmo** [comment]
Well ... not so fast 😆. Got the IntellTouch back up and running, with the HW mods as noted above - so all good there. But then I tried to reconnect my controller (RPi), only to find that it seems to have a blown USB port (ever heard of that?!?!). If I try to connect a USB thumbdrive (my OS) to one specific port, it flashes briefly, never comes up. Odd.

But - moved to another USB port (for now at least), OS comes back up. I re-connect my USB to RS-485, and it takes down the indoor controller. It's almost like the RS-485 adapter is blown? But that may have taken down the main board again? I can still use the buttons on the controller, but no joy with RS-485 control (or indoor controller). Arrgh.

So, I grabbed a Waveshare RPi RS-485 [HAT](https://www.waveshare.com/wiki/RS485_CAN_HAT). Seems to be happy (though the indoor controller still isn't 😞). That aside for a minute - has anyone every used a SPI interface to njsPC? Or for that matter, how to even echo the data as a start? Struggling there a bit. I did check, and in the kernel log,
```
[   12.408620] mcp251x spi0.0 can0: MCP2515 successfully initialized.
```
also,
```
$ lsmod | grep -i spi
spidev                 20480  0
spi_bcm2835            20480  0
```
and,
```
$ ls -l /dev/spi*
crw-rw---- 1 root spi 153, 0 Sep  7 18:14 /dev/spidev0.1
```
but,
```
$ sudo cat /dev/spidev0.1
cat: /dev/spidev0.1: Message too long
```

Any thoughts would be greatly appreciated!

**Reply by rstrouse** [reply]
What is the voltage on the RS485 port?  Could the regulated voltage (or current) be too high on the RS485 port?

What do you mean using SPI as an interface to njsPC.  Yes there is support in REM for the SPI0 and SPI1 using MCP chips as inputs but there aren't many device definitions created for that.  The only real devices that were created are temp sensors.  An RS485 software implementation is a fools errand considering the cheap availability of RS485 chips.

However, if you are referring to the Waveshare HAT using SPI, that is a device overlay at the OS level that should map to a tty device.  In that instance it will use one of the UART channels on the pi ttyAMA0 or ttyS0 for the primary hardware UART.   This device identifier is dependent on which Pi version you have.  Reading the fs for the low level spi device will not work as it is an unbuffered stream of infinite length.

**Reply by arrmo** [reply]
> However, if you are referring to the Waveshare HAT using SPI, that is a device overlay at the OS level that should map to a tty device. In that instance it will use one of the UART channels on the pi ttyAMA0 or ttyS0 for the primary hardware UART. This device identifier is dependent on which Pi version you have. Reading the fs for the low level spi device will not work as it is an unbuffered stream of infinite length.

Agreed! And that's what I was trying to figure out as well - fully agree with you, find the right tty. That's where I'm falling down ... LOL. Having issues finding that tty. ttyS0 doesn't exist, ttyAMA0 does. But not sure how to confirm that mapping. I need to dig more on that.

And a bit more debugging here. It seems like Tx from the OCP is "broken" => thinking the failed USB / RS-485 adapter (that I didn't know was fried, tried it again) has taken out a device on my mainboard (likely the 244, same device as before). Saying that because I do see traffic on the bus (found an old, somewhat broken, adapter to use for a minute) ... no broadcast messages, only messages (with the correct header) when I press a button on the ICP => and it works, turns on the spa light (for example), but no LED's on the ICP at all. And it won't turn back off afterwards from the ICP (but does from the OCP). So seems I need to (again) replace that 244, but also get the HAT working. Joy ... LMAO.

Thanks!

**Reply by rstrouse** [reply]
If this is a Raspberry Pi 3B then it will be ttyAMA0 and ttyS0 on a Pi4.   If you are on a Pi4 and there is no ttyS0 then it is likely that you do not have the Serial Port enabled in the raspi-config and/or the serial console is enabled.   Do you think a bad adapter caused the failure or is there runaway power on the bus from the OCP that is killing adapters?

**Reply by arrmo** [reply]
> If this is a Raspberry Pi 3B then it will be ttyAMA0 and ttyS0 on a Pi4.

Dang, not sure how you know that, but thanks! Pi 3B, so ttyAMA0.



> Do you think a bad adapter caused the failure or is there runaway power on the bus from the OCP that is killing adapters?

I think it's the adapter - it seems dead (others seem OK, even very old one). But that's also why I'm trying to get the hat working ... replacement, and verify my thinking. I have it connected now, stty to raw, but it's only spitting out 8 bytes then exiting (the od command). Will dig on that one, to get it streaming like USB0 does.

Thanks!

**Reply by rstrouse** [reply]
Make sure you do have the serial port enabled and the serial console enabled.  On pi3 the os may have also been hooked by Bluetooth.

**Reply by arrmo** [reply]
> Make sure you do have the serial port enabled and the serial console enabled. On pi3 the os may have also been hooked by Bluetooth.

Yep, just finding that in parallel ... LMAO! Thanks!

**Reply by rstrouse** [reply]
I meant disabled for the serial console.

**Comment by arrmo** [comment]
OK, after I enabled uart (config.txt), I had a "new" serial port, `/dev/ttyS0`. Woohoo! I then ran `od -x` ... data streaming 😄. But inverted it seemed, so I went out to swap the data wires (and on the way, noticed that my ICP is working again!). Swapped them ... now the ICP is down again, and no data from `od -x`. Arrgh! Will fight with it tomorrow 🤣.

Thanks!

**Reply by arrmo** [reply]
Oh, and I have a slight permissions issue 😜. `/dev/ttyAMA0` is dialout (group), but `/dev/ttyS0` is tty. Thinking I just need to add my user (pi) to `tty`.

**Reply by rstrouse** [reply]
There are two UARTs on a pi3 and depending on the config.txt the os maps ttyS0 to the software port then maps the ttyAMA0 to Bluetooth.  I don’t remember the exact settings but you will find some articles online on how to disable Bluetooth to get back the hardware UART.

**Reply by arrmo** [reply]
Yep, found that - once I get this working again I'll add it to the Wiki, share with other folks (FYI, it's `dtoverlay=disable-bt`). But as much as I said "tomorrow" I realized that the pin swap above really was wrong ... or rather, I did have it right initially. I swapped back, and the ICP is working again. Nice! But ... njsPC is still having issues, as is `od -x` => I'm thinking it's `stty` related, but I could swear I have that right. Still need to dig on that - as it's breaking both (and njsPC took 6+ hours to finish initializing, due to errors).

Thanks!


PS: I'll also capture in the Wiki info about masking, to make the serial port njsPC accessible.

**Reply by arrmo** [reply]
OK, 2 steps forward, 1 back ... LOL. Found some good info about the serial ports ([here](https://piers.rocks/raspberry/pi/rpi/serial/uart/pl011/16550/mini/zero/bcm2835/zerow/2019/02/18/raspberry-pi-serial-ports.html)), set mine up this way ... and I do seem to have the RS485 HAT (and njsPC) on ttyAMA0.

When I try `od -x` ... nothing, until I press a button on the ICP. OK, a bit odd, no broadcast, but that's OK (for a minute). I checked the output, and after some `stty` fumbling, I do see the header,
```
0000000 ff38 ffff ffff ffff 00ff a5ff 0f01 0210
0000020 0e1d 0024 0000 0000 0000 0000 5000 5e00
0000040 005e 5b00 0000 0400 0004 4d68 0100 3b03
0000060 ff3b ff00 01a5 1011 026e 0000 3701 ff37
0000100 ffff ffff ffff 00ff a5ff 0f01 0210 0e1d
0000120 0024 0000 0000 0000 0000 5000 5e00 005e
0000140 5b00 0000 0400 0004 4d68 0100 3b03 ff3b
0000160 ffff ffff ffff 00ff a5ff 0f01 0210 0e1d
0000200 0024 0000 0000 0000 0000 5000 5e00 005e
0000220 5b00 0000 0400 0004 4d68 0100 3b03 ff3b
0000240 ffff ffff ffff 00ff a5ff 0f01 0210 0e1d
0000260 0024 0000 0000 0000 0000 5000 5e00 005e
```

So, I would think all good. Of course not 😜. njsPC doesn't seem happy, not sure if it's an stty thing, or parity, flow control, etc. (or half duplex control), but lots of 
```
1|poolCont | RS485 Stats:{ "bytesReceived": 1056 "success": 21, "failed": 2, "bytesSent": 420, "collisions": 5, "failureRate": 8.70% }
1|poolCont | [08/09/2022, 14:33:51] info: [14:33:51] 192.168.2.77 GET /state/all?null {}
1|poolCont | [08/09/2022, 14:33:51] info: [14:33:51] 192.168.2.77 GET /state/all?null {}
1|poolCont | [08/09/2022, 14:33:52] warn: Message aborted after 4 attempt(s): 165,1,16,33,202,1,5,1,167
1|poolCont | [08/09/2022, 14:33:56] warn: Message aborted after 4 attempt(s): 165,1,16,33,202,1,6,1,168
1|poolCont | [08/09/2022, 14:34:00] warn: Message aborted after 4 attempt(s): 165,1,16,33,202,1,7,1,169
1|poolCont | [08/09/2022, 14:34:05] warn: Message aborted after 4 attempt(s): 165,1,16,33,202,1,8,1,170
```

Open to any suggestions! I'm not sure if there is a command line way to send, for example, pool or spa light on? Could use that to test at least.

Thanks!

**Reply by rstrouse** [reply]
If you are not receiving a broadcast status that means that the OCP is not sending out messages or the that message above is either njsPC or the ICP putting output on the bus.  If the OCP isn't sending back confirmation you will see the message aborted warning.

There can only be one of two things going on here either the OCP is not actually broadcasting or the RS485 on the pi is not receiving responses or broadcasts from the OCP.  If the above byte dump is from the ICP then that means that njsPC is receiving just fine but your OCP is not talking back.

If you load up the messageManager from the wiki you should be able to see all traffic from the RS485 port.
https://github.com/rstrouse/nodejs-poolController-dashPanel/wiki/Message-Manager

**Reply by tagyoureit** [reply]
Food for thought, when I had my EasyTouch go dead I had all sorts of comms challenges.  I shut off the OCP and set it up for Nixie to control my pump(s) directly.  All the wiring was still valid (you could disconnect the wires to the OCP to make sure) but that was enough that the fried chips on the EasyTouch board weren't interfering with the RS485 comms.  I was able to control the pumps without issue meaning that the OCP was definitely the culprit and had fried comms chips.  You might have blown one leg (the receive side) for example.

**Reply by arrmo** [reply]
Could be! A bit odd here, some more debugging 😄.
1. It really seems like USB in my RPi is fried. 3 of 4 ports "work", but if I try a USB-RS485 adapter (tried 3 of them!), the ICP doesn't work. Remove the device, ICP works great ... and all temps, controls, etc. So also thinking the OCP is OK?
2. Reconnected to the hat, ICP still working. Whew 😆. Then did `od -x`, pressed some buttons, watched ... yep, getting the correct header.
3. Started njsPC, `od -x` still in a second shell. OK for a bit, then loses it - data no longer looking correct, the header not there.

Not sure how to get back from 3 now ... LMAO! I have tried a couple attempts to reset the serial port (`/dev/ttyAMA0`), but no joy. I will reboot next, first need to stop njsPC from coming up at boot (I have it set to run, in pm2).

Thanks!

**Reply by tagyoureit** [reply]
You can't run 'od -x' and njsPC. Only one app can be accessing the port at a time.  But interesting tests to say the least.

**Reply by tagyoureit** [reply]
Also 'pm2 stop all' and 'pm2 save' should do it.

**Reply by arrmo** [reply]
> You can't run 'od -x' and njsPC. Only one app can be accessing the port at a time. But interesting tests to say the least.

Good to know 😆. Now to get things back - power cycled the RPi, also the OCP -> still not back! ICP shows correct information (and LED's), but controls not fully working. So screwy!



> Also 'pm2 stop all' and 'pm2 save' should do it.

Yep. I did `pm2 unstartup` for now.

**Reply by arrmo** [reply]
I think you're right @tagyoureit, next up - try to replace the RS-485 transceiver on the main board 😆 

Thanks!

**Comment by arrmo** [comment]
FYI, RS485 transceiver is a MAX1483ESA ... on order 😄

**Comment by arrmo** [comment]
OK, to circle back - close the loop, and share learnings that may help others. And noting ref designators where I remembered to capture them ... LOL.

The HW failures were a bit of a chain reaction, what I needed to replace,
1. Octal line driver / buffer ([Digi-Key](https://www.digikey.com/en/products/detail/toshiba-semiconductor-and-storage/74HC244D/6198937)): this was the part I could see had gone. In my case, it was U10 (there are a bunch of them on the board).
2. L1, series inductor feeding the LDO regulator. Regulator itself was fine ([Mouser](https://www.mouser.com/c/?q=LM1084IS-5.0%2FNOPB), in case needed), L1 fused when U10 failed.

Once the above was replaced, the main board was back up and running, and I could control the pool from the OCP. But ... not so fast, that wasn't all (of course 😆). RS-485 still wasn't working. Turns out due to the issues above (and voltage transients), or perhaps just due to time and or lightning (common here), my RS-485 link was "challenged". I had to replace the RS-485 transceiver ([Digi-Key](https://www.digikey.com/en/products/detail/analog-devices-inc-maxim-integrated/MAX1483ESA-T/1475845)) in the OCP ... and ICP both! A couple notes there as well,
1. The transceiver on the main board (OCP) is U4. Forgot to record the ref designator on the ICP, sorry!
2. There are TVS diodes in front of the transceiver. Mine were fine, but good to check them if you have issues. A [part I found](https://www.digikey.com/en/products/detail/littelfuse-inc/SMAJ12CA/762271), that is used on the RS485 hat noted below. Clamp seems about right for RS485.
3. The termination and bias network in front of the transceiver sets the bias to ~ 2.7 and 2.2V, to the actual SOIC inputs. Good to check this as well, if any problems. On the OCP, this is R27 (+5V on the lower side), R22 (GND on the lower side).
4. The HW issues also fried 2 of my USB-RS485 adapters, that made debugging a bit more "fun". I tracked this down by running the adapters back-to-back on a PC (Windows or Linux will work). Pitched the failed ones, and that helped a lot in debugging 🤣. BTW, this little [RPi hat](https://www.waveshare.com/wiki/RS485_CAN_HAT) works great, and it's quite cheap.

Hope this helps others! Oh, and BTW, when debugging the ICP, I connected my "roaming" (debug) RPi in parallel, so I had at the time 2 njsPC controllers on the RS485 bus. It all ran just fine 😄. But it's how I figured out that the ICP RS485 transceiver had gone also (i.e. the RPi was happy, connected to the same cabling).

Oh, and grand total ... cost me about $30, and a few gray hairs - a far cry from the $4900 the pool repair folks quoted me. Dang!

Thanks!

PS: Please don't take any of the links above as a "sales" job. I don't care where anyone buys their parts, just thought the info may help, that's all.

**Reply by rstrouse** [reply]
So glad you got it fixed.  Are you a hardware guy by trade or are you a dabbler like me?

**Reply by arrmo** [reply]
Thanks! By trade - have done it for ~ 32 years now (OK, that part hurts ... LOL). But definitely only a hacker on the SW side of things 😆.

**Reply by rstrouse** [reply]
Yeah I have the opposite going on regarding hardware and software. I've built professional applications in more languages than I can name.  So I guess I mostly flip the bits that others make.

**Reply by arrmo** [reply]
Ya, so you can blame us HW folks, we'll blame SW ... 😜. Situation normal.

FYI, my background is more RF, but I know enough BB to be dangerous 😆. If you do have any questions, don't hesitate to poke. Will definitely help if I can.

**Reply by rstrouse** [reply]
I sometimes cobble together a circuit board or two but its hard to get the propeller on my hat to spin the wrong way.

**Reply by arrmo** [reply]
Wrong way? I thought the HW one spun the right way? LMAO!

**Comment by tagyoureit** [comment]
I mentioned before I have a "dead" Easytouch right now.  I also believe my Intellichlor surge board to have a bad RS-485 chip.  I would be really interested in trying to tackle this but I'm not even sure where to start right now!

How hard is it for a layman to trace this stuff out, pull off the bad components, and refloat the new ones?  Do I need anything beyond a standard soldering iron and usual associated tools?

Super impressed you got it back working!!!

**Reply by arrmo** [reply]
Nothing super special. Do you have a picture of your RS485 transceiver? Mainly - how accessible is it?

Solder wick is always handy - and I'll be honest, to remove an SOIC ... 2 soldering irons is always easiest. I can explain if you want, NP at all!

First check the bias voltages, and also put some sort of RS485 adapter in parallel (for debug ... but depends on "how" the transceiver has failed).

**Reply by arrmo** [reply]
BTW, was talking to a guy at work today, and he mentioned the LTC2862 ... more robust, built in protection. FYI,
https://www.soselectronic.com/articles/linear-technology/ltc2862-the-next-step-towards-to-the-perfect-rs485-transceiver-1899

**Reply by tagyoureit** [reply]
Good to know... I'm up in San Francisco this week.  I'll try to pull the board(s) out and get some pics this weekend.  

I am hoping I just friend the RS485 chips.  Everything else on the OCP seems good but it won't get connected.  Same on the intellichlor surge board - I can' get consistent comms with my chlorinator.

**Reply by arrmo** [reply]
Sounds good, and sounds a lot like what I had happen. Fire me the pics if you want, I can try to help out (remotely ... LOL).

Do you have a USB adapter (or hat)? You can connect it in parallel, to see if the wiring is OK.

**Reply by tagyoureit** [reply]
I started a new [thread](https://github.com/tagyoureit/nodejs-poolController/discussions/619).  Let's see if you can help get me back up and running!  🤩

**Reply by arrmo** [reply]
All right, here we go 😆

**Comment by arrmo** [comment]
Do we need to report the spam here to GitHub?

**Reply by tagyoureit** [reply]
Yes, I have reported a few of them already... Are you sure this one is spam??  Hard to tell in another language!  😉

**Reply by arrmo** [reply]
LOL - true 😆

**Reply by rstrouse** [reply]
No kidding.  Chicks talk like that to me all the time.  I feel like a piece of meat sometimes.
